kubernetes/client/ws_client_test.py

```diff
@@ -32,6 +36,24 @@ def test_websocket_client(self):
                 ]:
             self.assertEqual(get_websocket_url(url), ws_url)
 
+    @patch.object(websocket._http, '_can_use_sni')
+    @patch.object(websocket._http, '_open_socket')
+    @patch.object(ssl.SSLContext, 'load_verify_locations')
+    def test_use_of_correct_ca_cert(self, load_verify_locations, *ignored):
+        # arrange
+        expected = '/expected-ca.pem'
+        configuration.ssl_ca_cert = expected
+        load_verify_locations.side_effect = Exception("need not go further")
+
+        # act
+        with self.assertRaises(Exception):
```

Comment:
If the code raises another exception, then it won't have reached the ssl context configuration step, and consequently the assertion will then fail. As is intended. I.e. The proper call to the ssl library wasn't made.

If however the code execution raises no exception until the step we're interested in, the call arguments are recorded in the mock, and we then raise an explicit exception to unwind back to the original call site in the unit test function. That's what I tried to convey with the exception message "go no further"; meaning don't execute a single additional statement in the library, since we have the info needed to determine expexted behavior at that point.

Not raising an exception would cause execution to continue in the websocket library, causing it to enter an event loop expecting socket related events, and hang the test.

In short, returning None would not cause a stack unwind (escape back to the original call site)